src/lib/circle.js

```diff
@@ -0,0 +1,179 @@
+
+const passwordStorage = require('./password-storage')('circleci');
+
+const circleConfig = {
+  version: 2,
+  jobs: {
+    build: {
+      docker: [
+        {
+          image: 'circleci/node:latest',
+        },
+      ],
+      steps: [
+        'checkout',
+        {
+          run: {
+            name: 'install',
+            command: 'npm install'
+          }
+        },
+        {
+          run: {
+            name: 'release',
+            command: 'npm run semantic-release || true',
+          },
+        },
+      ],
+    },
+  },
+};
+
+async function getUserInput(info) {
+  return await inquirer.prompt([
+    {
+      type: 'input',
+      name: 'token',
+      message: 'What is your CircleCI API token?',
+      validate: input => input.length === 40 ? true : 'Invalid token length',
+      default: async answers => {
+        const clipboardValue = await clipboard.read();
+        return clipboardValue.length === 40 ? clipboardValue : null
+      },
+      when: async answers => {
+        try {
+          const storedToken = await passwordStorage.get('token');
+          return !info.options.keychain || info.options['ask-for-passwords'] || !storedToken;
+        } catch (err) {
+          info.log.error('Something went wrong with your stored api token. Delete them from your keychain and try again');
+          process.exit(1)
+        }
+      }
+    },
+    {
+      type: 'confirm',
+      name: 'createConfigFile',
+      message: 'Do you want a `config.yml` file with semantic-release setup?',
+      default: true,
+    },
+    {
+      // TODO: Add step to existing config.yml
+      type: 'confirm',
+      name: 'overwrite',
+      default: false,
+      message: 'Do you want to overwrite the existing `config.yml`?',
+      when: answers => answers.createConfigFile && fs.existsSync('./.circleci/config.yml')
+    },
+  ]);
+}
+
+async function processToken(info)
+{
+  if (!info.circle.token) {
+    info.circle.token = await passwordStorage.get('token')
+  } else if (info.options.keychain) {
+    passwordStorage.set('token', info.circle.token);
+  }
+}
+
+async function setupCircleProject(info) {
+  const defaultReq = request.defaults({ 
+    headers: { 
+      'content-type': 'application/json',
+      'accept': 'application/json'
+    },
+    qs: {
+      'circle-token': info.circle.token
+    },
+    baseUrl: `https://circleci.com/api/v1.1/project/github/${info.ghrepo.slug[0]}/${info.ghrepo.slug[1]}/`,
+    json: true, 
+    simple: true 
+  })
+  
+  await followProject(info, defaultReq);
+  await addEnvironmentVariable(info, defaultReq, { name: 'GH_TOKEN', value: info.github.token });
+  await addEnvironmentVariable(info, defaultReq, { name: 'NPM_TOKEN', value: info.npm.token });
+}
+
+async function followProject(info, defaultReq) {
+  info.log.verbose(`Following repo ${info.ghrepo.slug[0]}/${info.ghrepo.slug[1]} on CircleCI...`)
+  const uri = `/follow`;
+  const success = await defaultReq.post(uri)
```

Comment:
When the request returns, is the synchronization repo/CircleCI is complete ?

We had several issue with Travis. The request would return but on Travis side the synchronization process wasn't complete and it wasn't possible to make the subsequent request (to add the env var) right away.

The problem would appear only on freshly created repo that are synchronized with CircleCI yet.

Reply:
I tested this with multiple repos that were not known before on CircleCI yet (so completely new). Whenever I got the first result back from CircleCI, I could always immediately add the env vars. There's no delay.

Reply:
Awesome. I mentioned by that was a big issue with Travis (and it currently is in some cases like #147)